dotnet/src/Agents/OpenAI/OpenAIAssistantChannel.cs

```diff
@@ -35,6 +35,12 @@ internal sealed class OpenAIAssistantChannel(AssistantsClient client, string thr
             RunStatus.Cancelled,
         ];
 
+    // Capture kernel function references for content processing
+    private sealed record KernelFunctionStep(string ToolCallId, KernelFunction Function, KernelArguments Arguments);
+
+    // Capture kernel function result for content processing
+    private sealed record KernelFunctionResult(string ToolCallId, FunctionResult Result);
```

Comment:
Consider using `FunctionCallContent` and `FunctionCallResult` classes instead of the records. I've quickly prototyped the change and it seems possible - https://github.com/microsoft/semantic-kernel/commit/eac0d514c2b27e021f1f1e528e1653507fa2b6bb

Reply:
Updated